p4-16/psa/PSA.mdk

```diff
@@ -971,22 +989,37 @@ its normal destination according to other features implemented by the
 P4 program, and in addition, send a copy of the packet as received to
 another output port, e.g. to a monitoring device.
 
-Packet cloning happens at the end of the ingress and/or egress pipeline. PSA
-specifies the following semantics for clone operation. When the clone
-operation is invoked at the end of the ingress pipeline, the cloned
-packet is a copy of the original packet from ingress port. When the
-clone operation is invoked at the end of egress pipeline, the cloned
-packet is a copy of the modified packet after egress processing. In
-both cases, the cloned packet is submitted to the egress pipeline for
-further processing.
+Packet cloning happens at the end of the ingress and/or egress
+pipeline. PSA specifies the following semantics for the clone
+operation.  When the clone operation is invoked at the end of the
+ingress pipeline, the cloned packet is a copy of the packet as it
+entered the ingress parser.  When the clone operation is invoked at
+the end of egress pipeline, the cloned packet is a copy of the
+modified packet after egress processing, as output by the egress
+deparser.  In both cases, the cloned packet is submitted to the egress
+pipeline for further processing.
+
+Logically, PRE implements the mechanics of copying a packet.  PRE
+reads the original packet as it began ingress parsing, or the modified
+packet from the egress deparser.  PRE then decides which version of
+the packet to use to generate a copy based on the value of the
+metadata provided by the ingress and egress pipeline.  The metadata
+fields that control cloning are those whose names begin with `clone`
+in types `psa_ingress_output_metadata_t` and
+`psa_egress_output_metadata_t`.
 
-Logically, PRE implements the mechanics of copying a packet. PRE reads
-the original packet from ingress port and the modified packet from
-egress deparser. PRE then decides which version of the packet to use
-to generate a copy based on the value of the metadata provided by the
-ingress and egress pipeline.  The metadata fields that control cloning
-are those whose names begin with `clone` in types
-`psa_ingress_output_metadata_t` and `psa_egress_output_metadata_t`.
+~ TBD
+The sentence beginning "PRE then decides which version of the packet
+to use to generate a copy" makes it sound like either you can create a
+clone of a packet in the ingress pipeline, or you can create a clone
+of a packet in the egress pipeline, but you cannot do both for the
+same packet.  The ingress and egress executions of packets, whether
+they occur due to the same original packet or different ones, are
+pretty much independent events, true?
```

Comment:
Correct, they are independent events and they need to be triggered as such. So maybe a rephrasing is required. How about removing that sentence and add after the "The metadata fields that control ..." sentence, the following: Based on the metadata settings the PRE will generate a copy of the packet using the correct version.

Reply:
I have tried simply eliminating the sentence in the commit I will add to this PR soon.  Give it a look over.  With the other text in the rest of this section, I don't see much that the sentence was saying that couldn't be understood from the rest.